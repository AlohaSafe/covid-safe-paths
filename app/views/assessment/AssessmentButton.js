import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';

import { Typography } from '../../components/Typography';
import Colors from '../../constants/colors';
import { Colors as AssessmentColors } from './constants';

/**
 * @typedef { import("react").ReactNode } ReactNode
 */

/** @type {React.FunctionComponent<{
 *   color?: string;
 *   onPress: () => void;
 *   title: string;
 *   disabled: boolean;
 * }>} */
const AssessmentButton = ({ color, onPress, title, disabled = false }) => {
  let backgroundColor = color
    ? color
    : (disabled
    ? Colors.GRAY_BUTTON
    : AssessmentColors.CTA);
  return (
    <TouchableOpacity
      onPress={onPress}
      disabled={disabled}
      testID='assessment-button'>
      <View
        style={[
          styles.cta,
          {
            backgroundColor: backgroundColor,
          },
        ]}>
        <Typography bold style={styles.ctaText}>
          {title}
        </Typography>
      </View>
    </TouchableOpacity>
  );
};

export default AssessmentButton;

const styles = StyleSheet.create({
  cta: {
    borderRadius: 12,
    padding: 14,
  },
  ctaText: {
    color: Colors.WHITE,
    //fontFamily: Fonts.primaryBold,
    fontSize: 18,
    textAlign: 'center',
  },
});
