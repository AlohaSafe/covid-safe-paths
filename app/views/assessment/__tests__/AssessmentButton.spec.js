import { render } from '@testing-library/react-native';
import React from 'react';

import AssessmentButton from '../AssessmentButton';

test('base', () => {
  const { asJSON } = render(<AssessmentButton title='Next' />);
  expect(asJSON()).toMatchSnapshot();
});

test('disabled', () => {
  const { asJSON } = render(<AssessmentButton disabled title='Next' />);
  expect(asJSON()).toMatchSnapshot();
});

test('color', () => {
  const { asJSON } = render(<AssessmentButton color='red' title='Next' />);
  expect(asJSON()).toMatchSnapshot();
});
