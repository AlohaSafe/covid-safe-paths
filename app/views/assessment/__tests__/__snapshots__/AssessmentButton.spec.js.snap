// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="assessment-button"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 12,
            "padding": 14,
          },
          Object {
            "backgroundColor": "rgb(0,121,253)",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "IBMPlexSans-Bold",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;

exports[`color 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="assessment-button"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 12,
            "padding": 14,
          },
          Object {
            "backgroundColor": "red",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "IBMPlexSans-Bold",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;

exports[`disabled 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="assessment-button"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 12,
            "padding": 14,
          },
          Object {
            "backgroundColor": "#868686",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "IBMPlexSans-Bold",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;
